core/domain/subtopic_page_domain.py

```diff
@@ -335,8 +271,27 @@ def update_page_contents_from_model(
         conversion_fn = getattr(
             cls, '_convert_page_contents_v%s_dict_to_v%s_dict' % (
                 current_version, current_version + 1))
-        versioned_page_contents['skill_contents'] = conversion_fn(
-            versioned_page_contents['skill_contents'])
+        versioned_page_contents['page_contents'] = conversion_fn(
+            versioned_page_contents['page_contents'])
+
+    @classmethod
+    def _convert_page_contents_v1_dict_to_v2_dict(cls, page_contents_dict):
+        """Converts from version 1 to 2. Version 2 replaces
```

Comment:
@aks681, I see another subtopic migration functionality in the topic_domain.py, I'm not sure whether I have to write something over there as well, can you please help?

Reply:
@aks681, can you please help me out?

More context: We have a one-off job for Topic migration but not for subtopic migration.

Reply:
Basically, there are 2 subtopic related models. One completely in the topic, for which the migration is initiated by `_migrate_subtopics_to_latest_schema` in topic_services.py and defined by `class subtopic` in topic_domain. I don't think your migration would affect that.
Then, there are subtopic pages which are linked to those subtopics by their IDs only. So, if you're migrating something exclusive to subtopic pages (which I think you are doing), it's fine to just add the migration here.

Reply:
@aks681 Thanks for the explanation, this will really help. One followup: SDo we have a one-off job for subtopic migration (like TopicMigrationManager)?

Reply:
@aks681, can you please reply to this?

Reply:
Oh, I don't think there is a subtopic_page one off job. I just searched and didn't find one. Do you need it asap? If so, I'll try to have a PR on for that.

Reply:
I think for this PR to get merged we need to have a complete migration structure, so it would be great if you can make a PR for the one-off job.

Reply:
Ok, I'll try to have a PR for that by the weekend.

Reply:
Hi @aks681, I'll be adding this job in this PR you can help me out with a review.

Reply:
Yeah, just ping me once the job is added!